### PR TITLE
Add a path for checking ZoneInformation without throwing an exception

### PR DESCRIPTION
**Pull request by powercode**
## PR Summary

In an effort to reduce the number of exceptions we throw as standard flowcontrol, this PR adds 
```
bool TryCreateFileStream(
    string path,
    string streamName, 
    FileMode mode, 
    FileAccess access, 
    FileShare share,  
   out FileStream stream) 
```

as a non-throwing alternative to `CreateFileStream`.

This makes the code slightly faster and debugging slightly less annoying.

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Change is not breaking](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` to the beginning of the title and remove the prefix when the PR is ready.
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [ ] User-facing [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed - Issue link:
- **Testing - New and feature**
    - [x] Not Applicable or can only be tested interactively
    - **OR**
    - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
        - [ ] [Add `[feature]` if the change is significant or affects feature tests](https://github.com/PowerShell/PowerShell/blob/master/docs/testing-guidelines/testing-guidelines.md#requesting-additional-tests-for-a-pr)

### CODE REVIEW

File: src/System.Management.Automation/namespaces/FileSystemProvider.cs [hunk 1]
```diff
@@ -8262,6 +8262,41 @@ internal static FileStream CreateFileStream(string path, string streamName, File
...
+        /// Tries to create a file stream on a file.
+        /// </summary>
+        /// <param name="path">The fully-qualified path to the file.</param>
+        /// <param name="streamName">The name of the alternate data stream to open.</param>
+        /// <param name="mode">The FileMode of the file.</param>
+        /// <param name="access">The FileAccess of the file.</param>
+        /// <param name="share">The FileShare of the file.</param>
+        /// <param name="stream">A FileStream that can be used to interact with the file.</param>
+        /// <returns>true if the stream was successfully created, otherwise false.</returns>
+        internal static bool TryCreateFileStream(string path, string streamName, FileMode mode, FileAccess access, FileShare share, out FileStream stream)
```
Comment (iSazonov) [inline src/System.Management.Automation/namespaces/FileSystemProvider.cs]: Both CreateFileStream and TryCreateFileStream use common code - I think we should use "dry" principle. 

Also please fix formatting issues for the code.

File: src/System.Management.Automation/namespaces/FileSystemProvider.cs [hunk 2]
```diff
@@ -8242,24 +8242,14 @@ internal static List<AlternateStreamData> GetStreams(string path)
...
-            if (mode == FileMode.Append) mode = FileMode.OpenOrCreate;
-            SafeFileHandle handle = NativeMethods.CreateFile(resultPath, access, share, IntPtr.Zero, mode, 0, IntPtr.Zero);
-
-            if (handle.IsInvalid)
+            if (!TryCreateFileStream(path, streamName, mode, access, share, out var stream))
             {
                 string errorMessage = StringUtil.Format(
                     FileSystemProviderStrings.AlternateDataStreamNotFound, streamName, path);
-                throw new FileNotFoundException(errorMessage, resultPath);
+                throw new FileNotFoundException(errorMessage, $"{path}:{streamName.Trim()}");
```
Comment (iSazonov) [inline src/System.Management.Automation/namespaces/FileSystemProvider.cs]: Do we really need `Trim()` for `streamName` in the _internal_ methods? We do not this for `path`.

Comment (powercode) [reply src/System.Management.Automation/namespaces/FileSystemProvider.cs]: Not sure - I just opted to not change any functionality. Small cost on the error path.

Comment (iSazonov) [reply src/System.Management.Automation/namespaces/FileSystemProvider.cs]: I vote to remove - I believe consumer should take care.

**Review by iSazonov — APPROVED** [review]

**Review by anmenaga — APPROVED** [review]

**Review by TravisEz13 — COMMENTED** [review]
LGTM

**Pull request merged**